### PR TITLE
Problem: cometbft is outdated

### PR DESCRIPTION
**Pull request by mmsqe**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **New Features**
	- Introduced a new configuration option for cache size, allowing users to control caching behavior for improved performance.
- **Dependency Updates**
	- Upgraded the CometBFT library to version `v0.38.11`, potentially enhancing functionality and fixing bugs.
	- Removed an unnecessary indirect dependency on `github.com/libp2p/go-buffer-pool`, streamlining project dependencies.

### CODE REVIEW

**Review by yihuang — APPROVED** [review]

File: integration_tests/configs/default.jsonnet [hunk 1]
```diff
@@ -42,6 +42,7 @@
           enable: true,
           'zero-copy': true,
           'snapshot-interval': 5,
+          'cache-size': 0,
```
Comment (yihuang) [inline integration_tests/configs/default.jsonnet:45]: `cache-size` is automatically `0` if block-stm is used.

Comment (mmsqe) [reply integration_tests/configs/default.jsonnet:45]: default is 1000 if not specify, might need change for testground as well

Comment (yihuang) [reply integration_tests/configs/default.jsonnet:45]: https://github.com/crypto-org-chain/cronos/pull/1547

Comment (mmsqe) [reply integration_tests/configs/default.jsonnet:45]: right, seems no other place need overwrite this config

**Pull request merged**